tests/integration/cartography/intel/github/test_repos.py

```diff
@@ -186,3 +186,27 @@ def test_repository_to_collaborators(neo4j_session):
     actual_nodes = {n['collab_count'] for n in nodes}
     expected_nodes = {5}
     assert actual_nodes == expected_nodes
+
+
+def test_library_to_repo(neo4j_session):
+    """
+    Ensure that repositories are connected to Python libraries
+    """
+    # Create the path (:Repo1)-[:REQUIRES]->(:PythonLibrary{'Cartography'})<-[:REQUIRES]-(:Repo1),
+    # and test that exactly 1 repo is connected to the PythonLibrary.
+    _ensure_local_neo4j_has_test_data(neo4j_session)
+
+    nodes = neo4j_session.run("""
+    MATCH (repo:GitHubRepository)-[:REQUIRES]->(lib:PythonLibrary{name:'cartography'})
+    RETURN lib.id
+    """)
+    print(list(n['lib.id'] for n in nodes))
+
+    query = """
+    MATCH (repo:GitHubRepository)-[:REQUIRES]->(lib:PythonLibrary{id:'cartography|0.1.0'})
```

Comment:
It would be good to have some tests that cover non-pinned dependencies and how our code handles them (e.g. adding an edge to a node with no version, and potentially recording the specifier on the edge).